adapters/33across/33across.go

```diff
@@ -38,39 +51,62 @@ func (a *TtxAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters
 	var errs []error
 	var adapterRequests []*adapters.RequestData
 
-	adapterReq, errors := a.makeRequest(request)
-	if adapterReq != nil {
-		adapterRequests = append(adapterRequests, adapterReq)
+	// Break up multi-imp request into multiple external requests since we don't
+	// support SRA in our exchange server
+	for i := 0; i < len(request.Imp); i++ {
+		adapterReq, err := a.makeRequest(request, request.Imp[i])
+		if adapterReq != nil {
+			adapterRequests = append(adapterRequests, adapterReq)
+		}
+		if err != nil {
+			errs = append(errs, err)
+		}
 	}
-	errs = append(errs, errors...)
 
-	return adapterRequests, errors
+	return adapterRequests, errs
 }
 
-// Update the request object to include custom value
-// site.id
-func (a *TtxAdapter) makeRequest(request *openrtb.BidRequest) (*adapters.RequestData, []error) {
-	var errs []error
+func (a *TtxAdapter) makeRequest(request *openrtb.BidRequest, imp openrtb.Imp) (*adapters.RequestData, error) {
+	var imps []openrtb.Imp
 
-	// Make a copy as we don't want to change the original request
 	reqCopy := *request
-	if err := preprocess(&reqCopy); err != nil {
-		errs = append(errs, err)
+	impCopy, err := makeImps(imp)
+
+	if err != nil {
+		return nil, err
 	}
 
-	if reqCopy.Imp[0].Banner == nil && reqCopy.Imp[0].Video == nil {
-		errs = append(errs, &errortypes.BadInput{
-			Message: "At least one of [banner, video] formats must be defined in Imp. None found",
-		})
+	reqCopy.Imp = append(imps, impCopy)
 
-		return nil, errs
+	// Add info about caller
+	caller := TtxCaller{"Prebid-Server", "n/a"}
```

Comment:
Consider making this a constant / static value to avoid memory allocations per call.

Reply:
I am new to Golang. How would I make this into a const? Declare it as a top-level variable?

Reply:
Yup. It can't literally be a const because it's a struct, but a package/top level variable which you only read from is how you would do it.